### PR TITLE
Meta: Specify required Node version in docs and in nvmrc

### LINKED ISSUE #3923: Struggling installing dependencies with Node v13, v14 and v15
I might only me but...

I was trying to install the dependencies of the project but I have weird results with each version of Node.js.

## When installing with Node.js version 13 or 14

I used to use the version 13 and it worked without any extra change.

Now each time I get the following errors in my terminal + the `package-lock.json` file is entirely changed:

<details>
<summary>Result in my terminal</summary>

```text
08:46:24 on  main via ⬢ v13.14.0
➜ npm i
npm WARN read-shrinkwrap This version of npm is compatible with lockfileVersion@1, but package-lock.json was generated for lockfileVersion@2. I'll try to do my best with it!

> dtrace-provider@0.8.8 install /Users/MY_USER/Dev/refined-github/main/node_modules/dtrace-provider
> node-gyp rebuild || node suppress-error.js

  ACTION binding_gyp_ndtp_target_build_ndtp .
-----------------------------------------------------------------
Building dtrace-provider failed with exit code 1, falling back
on stub implementation.

Re-run install with V set in your environment to see the build
output, or NODE_DTRACE_PROVIDER_REQUIRE=hard to force an
installation failure.
-----------------------------------------------------------------
  TOUCH Release/obj.target/ndtp.stamp

> fsevents@1.2.13 install /Users/MY_USER/Dev/refined-github/main/node_modules/watchpack-chokidar2/node_modules/fsevents
> node install.js

  SOLINK_MODULE(target) Release/.node
  CXX(target) Release/obj.target/fse/fsevents.o
  SOLINK_MODULE(target) Release/fse.node

> core-js@2.6.12 postinstall /Users/MY_USER/Dev/refined-github/main/node_modules/core-js
> node -e "try{require('./postinstall')}catch(e){}"

Thank you for using core-js ( https://github.com/zloirock/core-js ) for polyfilling JavaScript standard library!

The project needs your help! Please consider supporting of core-js on Open Collective or Patreon:
> https://opencollective.com/core-js
> https://www.patreon.com/zloirock

Also, the author of core-js ( https://github.com/zloirock ) is looking for a good job -)


> esbuild@0.8.28 postinstall /Users/MY_USER/Dev/refined-github/main/node_modules/esbuild
> node install.js


> core-js@3.6.5 postinstall /Users/MY_USER/Dev/refined-github/main/node_modules/sign-addon/node_modules/core-js
> node -e "try{require('./postinstall')}catch(e){}"


> spawn-sync@1.0.15 postinstall /Users/MY_USER/Dev/refined-github/main/node_modules/spawn-sync
> node postinstall

added 1644 packages from 859 contributors and audited 1652 packages in 34.054s

174 packages are looking for funding
  run `npm fund` for details

found 1 high severity vulnerability
  run `npm audit fix` to fix them, or `npm audit` for details

08:47:02 on  main [!] via ⬢ v13.14.0 took 36s
```
</details>

## When installing with Node.js version 15

Seems better, no more error in the terminal (see the result below) but the `package-lock.json` get its field `name` changed to `main`, see:

gh-refined

<details>
<summary>Result in my terminal</summary>

```text
08:49:59 on  main via ⬢ v15.7.0
➜ npm i
npm WARN deprecated request-promise-native@1.0.9: request-promise-native has been deprecated because it extends the now deprecated request package, see https://github.com/request/request/issues/3142
npm WARN deprecated urix@0.1.0: Please see https://github.com/lydell/urix#deprecated
npm WARN deprecated har-validator@5.1.5: this library is no longer supported
npm WARN deprecated fsevents@2.1.3: Please update to v 2.2.x
npm WARN deprecated resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated
npm WARN deprecated jetpack-id@1.0.0: Jetpack extensions are no longer supported by Firefox. Use web-ext instead to build a WebExtension.
npm WARN deprecated fsevents@1.2.13: fsevents 1 will break on node v14+ and could be using insecure binaries. Upgrade to fsevents 2.
npm WARN deprecated chokidar@2.1.8: Chokidar 2 will break on node v14+. Upgrade to chokidar 3 with 15x less dependencies.
npm WARN deprecated js-select@0.6.0: Package no longer supported. Contact support@npmjs.com for more info.
npm WARN deprecated request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142
npm WARN deprecated @babel/polyfill@7.12.1: 🚨 This package has been deprecated in favor of separate inclusion of a polyfill and regenerator-runtime (when needed). See the @babel/polyfill docs (https://babeljs.io/docs/en/babel-polyfill) for more information.
npm WARN deprecated axios@0.19.2: Critical security vulnerability fixed in v0.21.1. For more information, see https://github.com/axios/axios/pull/3410
npm WARN deprecated core-js@2.6.12: core-js@<3 is no longer maintained and not recommended for usage due to the number of issues. Please, upgrade your dependencies to the actual version of core-js@3.

added 1772 packages, and audited 1773 packages in 17s

175 packages are looking for funding
  run `npm fund` for details

2 high severity vulnerabilities

To address all issues (including breaking changes), run:
  npm audit fix --force

Run `npm audit` for details.

08:50:19 on  main [!] via ⬢ v15.7.0 took 17s
```
</details>

It seems Node.js v15 should be preferred now but I would like a confirmation from the maintainers since you still mention "Node.js v13 or later" in your contributing file:
https://github.com/sindresorhus/refined-github/blob/49971cd4c32a73171fc1c028c74bc22a014ce71d/contributing.md#L82-L84

Thank you.

### PR DESCRIPTION
**Pull request by MrChocolatine**
1. LINKED ISSUES:
   Closes #3923 

2. TEST URLS:
   x

3. SCREENSHOT:
   x

Update the required Node.js version in the contributing file.

### CODE REVIEW

**Comment by busches** [comment]
Thoughts on adding a `.nvmrc` file to make this automated for most?

**Comment by MrChocolatine** [comment]
@busches I thought about that but I suppose if we start adding such a file we should also add files for other node managers...

**Comment by MrChocolatine** [comment]
However we could add the entry [`engines.node`](https://docs.npmjs.com/cli/v6/configuring-npm/package-json/#engines) to `package.json`.

**Comment by fregante** [comment]
As far as I know `engine` doesn’t apply to the local project, but only to dependencies. 

As for nvmrc, people have to notice it and manually call `nvm use`, it’s not automatic. So it would be one more thing to keep up to date. 

There’s no easy way to prevent this before npm install… unless… is the `preinstall` script called when installing the project? 🤔

**Comment by busches** [comment]
I guess I have my she'll configured to call nvm use when I change directory, but not everyone will be like that. Either way, having a more obvious way to know which node/npm work with the project would be ideal.

File: contributing.md [hunk 1]
```diff
@@ -81,7 +81,7 @@ void features.add(__filebasename, {
 
 ## Requirements
 
-[Node.js](https://nodejs.org/en/download/) version 13 or later is required.
+[Node.js](https://nodejs.org/en/download/) version 15 is required.
```
Comment (fregante) [inline contributing.md:84]:
```suggestion
[Node.js](https://nodejs.org/en/download/) version 15 or later is required. <!-- Also update .nvmrc -->
```

Let's add an .nvmrc file, at least it's useful to some. I noticed that configuring `cd` to also execute that file is a common pattern.

**Comment by MrChocolatine** [comment]
> I guess I have my shell configured to call nvm use when I change directory, but not everyone will be like that. Either way, having a more obvious way to know which node/npm work with the project would be ideal.

Yes, it is not automatic, you must be probably referring to this section:
https://github.com/nvm-sh/nvm#deeper-shell-integration

@fregante, what about similar files for other Node.js versions managers? (I mean I don't mind at all, I use `nvm` but why `.nvmrc` should prevail over the others?)
- https://github.com/tj/n#specifying-node-versions
- https://github.com/shadowspawn/node-version-usage
- https://docs.volta.sh/guide/

Also, [in this section](https://github.com/tj/n#specifying-node-versions) we can read:
> The `engine` label looks for a `package.json` file and reads the `engines` field to determine compatible `node`. Requires an installed version of `node`, and uses `npx semver` to resolve complex ranges.
Should we take this in condideration?


Regarding the `README`, should we add something about this `.nvmrc` file? Like:
> If you use `nvm`, you can use the command `nvm use` to automatically use the required version of Node.js

**Comment by MrChocolatine** [comment]
For now I put `15` in the `.nvmrc` file as I am not sure `nvm` can actually handle a string like `>= 15`.

**Comment by MrChocolatine** [comment]
No answer for the previous questions?

**Pull request merged**